Tasks/AzureRmWebAppDeployment/deploymentProvider/PublishProfileWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,38 @@
+import { IWebAppDeploymentProvider } from "./IWebAppDeploymentProvider";
+import { TaskParameters } from "../operations/TaskParameters";
+import {PublishProfileUtility} from '../operations/PublishProfileUtility';
+import { FileTransformsUtility } from '../operations/FileTransformsUtility';
+import tl = require('vsts-task-lib/task');
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var deployUtility = require('webdeployment-common/utility.js');
+var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
+
+export class PublishProfileWebAppDeploymentProvider implements IWebAppDeploymentProvider{
+    private taskParams: TaskParameters;
+
+    constructor(taskParams: TaskParameters) {
+        this.taskParams = taskParams;
+    }
+
+    public async PreDeploymentStep(){ }
```

Comment:
we might want to set the output variable for publish profile method too.
We can use the value from `SiteUrlToLaunchAfterPublish` attribute.
![image](https://user-images.githubusercontent.com/20121508/38239922-c5304648-374b-11e8-8314-559d677e461b.png)

Reply:
nit: we can use this value to ping the application to avoid `ERROR_CONNECTION_TERMINATED`.
Adding @Ajay-MS  to comment.